packages/toolpad-app/src/toolpadDataSources/rest/client.tsx

```diff
@@ -363,10 +363,10 @@ function QueryEditor({
     () => ({
       parameters: previewParams,
       process: {
-        env,
+        env: Object.fromEntries(envVarNames.map((varName) => [varName, '<SECRET>'])),
```

Comment:
Oh, we're evaluating some bindings on the client side...
Then I guess it isn't possible to only send the variable names

Reply:
So it's not secure this way? I thought it only uses the real environment values if it runs server-side :/ Shouldn't that be possible?

Reply:
Ah, yes client-side it won't use the real values in this case, yeah...

Reply:
I can show `<SECRET>` and use the real values in the client anyway if that's still safe.

Reply:
I think we can do either one of those:

* Send the whole env like you did before and let it be
* Rethink how to use secrets in rest datasource and instead offer a new type of binding, next to javascript expression, allow binding directly to an environment variable through a dropdown

Reply:
I feel like we can go with the first option "Send the whole env like you did before and let it be".
But let me know if you think the second one is better for any reason and it would not be that much extra work anyway.

Reply:
Some arguments I can think of that would be in favor for the second solution:

- It's easier for us to detect which environment variables an app expects to be run under. That allows us to warn on any missing ones, e.g. when a user checks out an app for the first time. It could somewhat be extracted from the javascript bindings by regex, or static analysis, but it wouldn't be a foolproof solution.
- We could more fine grained specify where exactly we want this to be used. e.g. only in authentication/headers
- Thinking long term, suppose we decide that whatever we came up with in this PR was a mistake, we can migrate from `$env: MY_VAR` to `$jsExpression: 'process.env.MY_var'` and guarantee no breaking change, we can't do that on the inverse.

Reply:
Fair points, ok I'll change it to be a new type of binding, having it as `$env` sounds consistent with the other types.